File: app/src/main/java/com/song/deviceinfo/ui/maps/MapsFragment.java
package com.song.deviceinfo.ui.maps;

import com.song.deviceinfo.ui.base.BaseViewModel;
import com.song.deviceinfo.ui.oneline.OneLineFragment;
import com.song.deviceinfo.utils.LogUtils;
import com.song.deviceinfo.utils.ThreadPoolUtils;

import org.json.JSONArray;

import java.util.List;

import androidx.lifecycle.ViewModelProviders;

/**
 * Created by chensongsong on 2021/9/10.
 */
public class MapsFragment extends OneLineFragment {

    @Override
    protected BaseViewModel crateViewModel() {
        return ViewModelProviders.of(this).get(MapsViewModel.class);
    }

    @Override
    protected void refreshData() {
        ThreadPoolUtils.getInstance().execute(() -> {
            final List<String> list = getMapsInfo();
            JSONArray jsonArray = new JSONArray();
            for (String item : list) {
                try {
                    jsonArray.put(item);
                } catch (Exception e) {
                    e.printStackTrace();
                }
            }
            LogUtils.printLongString(jsonArray.toString());
            mainHandler.post(() -> {
                viewModel.setValue(list);
            });
        });
    }

    private List<String> getMapsInfo() {
        return ((MapsViewModel) viewModel).getMapsInfo(getContext());
    }

}

File: app/src/main/java/com/song/deviceinfo/ui/maps/MapsFragment.kt
package com.song.deviceinfo.ui.maps

import androidx.lifecycle.ViewModelProvider
import com.song.deviceinfo.ui.base.BaseViewModel
import com.song.deviceinfo.ui.oneline.OneLineFragment
import com.song.deviceinfo.utils.LogUtils.printLongString
import org.json.JSONArray

/**
 * Created by chensongsong on 2021/9/10.
 */
class MapsFragment : OneLineFragment() {
    override fun createViewModel(): BaseViewModel<String> {
        return ViewModelProvider(this)[MapsViewModel::class.java]
    }
    
    override fun refreshData() {
        launchOnIO {
            val list = mapsInfo
            val jsonArray = JSONArray()
            for (item in list) {
                try {
                    jsonArray.put(item)
                } catch (e: Exception) {
                    e.printStackTrace()
                }
            }
            printLongString(jsonArray.toString())
            launchOnMain { viewModel?.setItems(list) }
        }
    }
    
    private val mapsInfo: List<String>
        get() = (viewModel as MapsViewModel).getMapsInfo(context)
}
